### PR TITLE
chore: Add merchant specific config

### PR DESCRIPTION
**Pull request by JeevaRamu0104**
## Type of Change

- [ ] Bugfix
- [ ] New feature
- [x] Enhancement
- [ ] Refactoring
- [ ] Dependency updates
- [ ] Documentation
- [ ] CI/CD

## Description

Added the merchant specific config.Set the required ids in the env
```
export default__merchant_config__new_analytics__org_ids="103,104";
export default__merchant_config__new_analytics__merchant_ids="203,204";
export default__merchant_config__new_analytics__profile_ids="303,304";
```
image

## Motivation and Context
To display a feature specific to a merchant.

## How did you test it?
Add the respective ids in the autopilot 

## Where to test it?

- [x] INTEG
- [x] SANDBOX
- [ ] PROD

## Checklist

- [ ] I ran `npm run re:build`
- [ ] I reviewed submitted code
- [ ] I added unit tests for my changes where possible

### CODE REVIEW

**Review by sagarnaikjuspay — APPROVED** [review]

**Review by gitanjli525 — DISMISSED** [review]

File: src/screens/Hooks/MerchantSpecificConfigHook.res [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+    try {
+      let domain = HyperSwitchEntryUtils.getSessionData(~key="domain", ~defaultValue="default")
+      let merchantConfigURL = ` ${GlobalVars.getHostUrlWithBasePath}/config/merchant?domain=${domain}`
+      let body =
+        [
+          ("org_id", orgId->JSON.Encode.string),
+          ("merchant_id", merchantId->JSON.Encode.string),
+          ("profile_id", profileId->JSON.Encode.string),
+        ]
+        ->Dict.fromArray
```
Comment (PritishBudhiraja) [inline src/screens/Hooks/MerchantSpecificConfigHook.res:40]: getDictFromJson some func

File: src/server/config.mjs [hunk 2]
```diff
@@ -23,6 +23,70 @@ function updateConfigWithEnv(updatedConfig, domain = "", prefix = "") {
...
+  } else if (
+    tomlConfig &&
+    tomlConfig !== null &&
+    typeof tomlConfig === "object"
+  ) {
+    return tomlConfig;
+  } else {
+    return [];
+  }
+}
```
Comment (PritishBudhiraja) [inline src/server/config.mjs:48]:
```suggestion
function checkEnvValues(env, tomlConfig) {
  if (typeof env === "string") {
    const data = env.split(",");
    return Array.isArray(data) ? data : [];
  }

  if (typeof tomlConfig === "object" && tomlConfig !== null) {
    return tomlConfig;
  }

  return [];
}
```

File: src/screens/Hooks/MerchantSpecificConfigHook.res (hunk 1 above)
Comment (JeevaRamu0104) [reply src/screens/Hooks/MerchantSpecificConfigHook.res:40]: there is no function

Comment (PritishBudhiraja) [reply src/screens/Hooks/MerchantSpecificConfigHook.res:40]: getJsonFromArrayOfJson sorry this one

**Review by PritishBudhiraja — APPROVED** [review]

File: src/screens/Hooks/MerchantSpecificConfigHook.res [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+        let err = Exn.message(e)->Option.getOr("Failed to Fetch!")
+        Exn.raiseError(err)
+      }
+    }
+  }
+  let useHasEnabledForMerchant = (config: FeatureFlagUtils.config) => {
+    config.orgIds->Array.length > 0 ||
+    config.merchantIds->Array.length > 0 ||
+    config.profileIds->Array.length > 0
+  }
```
Comment (Riddhiagrawal001) [inline src/screens/Hooks/MerchantSpecificConfigHook.res:56]: isFeatureEnabledForMerchant

**Review by Riddhiagrawal001 — APPROVED** [review]

File: src/screens/Hooks/MerchantSpecificConfigHook.res (hunk 1 above)
Comment (JeevaRamu0104) [reply src/screens/Hooks/MerchantSpecificConfigHook.res]: updated

**Review by PritishBudhiraja — APPROVED** [review]

**Review by Riddhiagrawal001 — APPROVED** [review]

**Pull request merged**